cupyx/scipy/fftpack/fft.py

```diff
@@ -15,52 +15,86 @@ def get_fft_plan(a, shape=None, axes=None, value_type='C2C'):
             output. If ``shape`` is not given, the lengths of the input along
             the axes specified by ``axes`` are used.
         axes (None or int or tuple of int):  The axes of the array to
-            transform. Currently, these must be a set of up to three adjacent
-            axes and must include either the first or the last axis of the
-            array.  If `None`, it is assumed that all axes are transformed.
+            transform. If `None`, it is assumed that all axes are transformed.
+
+            Currently, for performing N-D transform these must be a set of up
+            to three adjacent axes, and must include either the first or the
+            last axis of the array.
         value_type ('C2C'): The FFT type to perform.
             Currently only complex-to-complex transforms are supported.
 
     Returns:
-        plan (cupy.cuda.cufft.PlanNd): The cuFFT Plan.
+        plan: a cuFFT plan for either 1D transform (cupy.cuda.cufft.Plan1d)
+            or N-D transform (cupy.cuda.cufft.PlanNd).
     """
+    # check value_type
     fft_type = _convert_fft_type(a, value_type)
     if fft_type not in [cufft.CUFFT_C2C, cufft.CUFFT_Z2Z]:
         raise NotImplementedError("Only C2C and Z2Z are supported.")
 
+    # check input array
     if a.flags.c_contiguous:
         order = 'C'
     elif a.flags.f_contiguous:
         order = 'F'
     else:
         raise ValueError("Input array a must be contiguous")
 
-    if (shape is not None) and (axes is not None) and len(shape) != len(axes):
-        raise ValueError("Shape and axes have different lengths.")
-    if (shape is not None) and (axes is None) and len(shape) != a.ndim:
+    # check axes
+    # n=1: 1d (need axis1D); n>1: Nd
+    if axes is None:
+        n = a.ndim
+        axes = tuple(i for i in range(n))
+        if n == 1:
+            axis1D = 0
+    elif isinstance(axes, int):
+        n = 1
+        axis1D = axes
+        axes = (axes,)
+        if axis1D >= a.ndim or axis1D < -a.ndim:
+            raise ValueError("Invalid input: a has dimension {0} but you "
+                             "choose the axis {1} to be transformed."
```

Comment:
I don't think it is necessary to say "Invalid input: " as the ValueError itself indicates this.  I would just simplify to something like:
```python
"The chosen axis ({0}) exceeds the number of dimensions of a ({1})".format(axis1D, a.ndim)
```

Reply:
Good point, done.